lang/src/org/partiql/lang/planner/transforms/LogicalToLogicalResolvedVisitorTransform.kt

```diff
@@ -195,6 +198,22 @@ internal data class LogicalToLogicalResolvedVisitorTransform(
                 locals = emptyList(), // NOTE: locals will be populated by caller
                 metas = node.metas
             )
+        }.apply {
+            /**
+             * Quick validation step to make sure the indexes of any variables make sense.
+             * It is unlikely that this check will ever fail, but if it does, it likely means there's a bug in
+             * [org.partiql.lang.planner.transforms.VariableIdAllocator] or that the plan was malformed by other means.
+             */
+            locals.forEachIndexed { idx, it ->
+                if (it.registerIndex.value != idx.toLong()) {
+                    throw EvaluationException(
+                        message = "Variable index must match ordinal position of variable",
+                        errorCode = ErrorCode.INTERNAL_ERROR,
+                        errorContext = propertyValueMapOf(),
+                        internal = true
+                    )
+                }
+            }
```

Comment:
again, validation doesn't belong in transform passes. Were these changes meant to be included?

Reply:
I agree that validation should have its own pass. But do you think we really need the check here?

Reply:
It is trying to ensure the code we are writing is correct